integration_tests/dkim_test.go

```diff
@@ -0,0 +1,163 @@
+const pubKey_2 = "MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAv3bzh5rabT+IWegVAoGnS/kRO2kbgr+jls+Gm5S/bsYYCS/MFsWBuegRE8yHwfiyT5Q90KzwZGkeGL609yrgZKJDHv4TM2kmybi4Kr/CsnhjVojMM7iZVu2Ncx/i/PaCEJzo94dcd4nIS+GXrFnRxU/vIilLojJ01W+jwuxrrkNg8zx6a9wWRwdQUYGUIbGkYazPdYUd/8M8rviLwT9qsnJcM4b3Ie/gtcYzsL5LhuvhfbhRVNGXEMADasx++xxfbIpPr5AgpnZo+6rA1UCUfwZT83Q2pAybaOcpjGUEWpP8h30Gi5xiUBR8rLjweG3MtYlnqTHSyiHGUt9JSCXGPQIDAQAB"
+
+const domain_1 = "x.com"
+const domain_2 = "xion.com"
+
+const selector_1 = "dkim202406"
+const selector_2 = "dkim202407"
+
+const poseidon_hash_1 = "1983664618407009423875829639306275185491946247764487749439145140682408188330"
+const poseidon_hash_2 = "1983664618407009423875829639306275185491946247764487749439145140682408188330"
+
+const customDomain = "account.netflix.com"
+const customSelector = "kk6c473czcop4fqv6yhfgiqupmfz3cm2"
+
+var pubKeysBz, _ = json.Marshal([]Dkim{{
+	PubKey:       pubKey_1,
+	Domain:       domain_1,
+	Selector:     selector_1,
+	PoseidonHash: poseidon_hash_1,
+}, {
+	PubKey:       pubKey_2,
+	Domain:       domain_2,
+	Selector:     selector_2,
+	PoseidonHash: poseidon_hash_2,
+}})
+
+func TestDKIMModule(t *testing.T) {
+	td := BuildXionChain(t, "0.0uxion", ModifyInterChainGenesis(ModifyInterChainGenesisFn{ModifyGenesisDKIMRecords, ModifyGenesisShortProposals}, [][]string{{string(pubKeysBz)}, {votingPeriod, maxDepositPeriod}}))
+
+	xion, ctx := td.xionChain, td.ctx
+
+	config := sdk.GetConfig()
+	config.SetBech32PrefixForAccount("xion", "xionpub")
+	xion.Config().EncodingConfig.InterfaceRegistry.RegisterImplementations((*sdk.Msg)(nil), &dkimTypes.MsgAddDkimPubKeys{})
+	xion.Config().EncodingConfig.InterfaceRegistry.RegisterImplementations((*sdk.Msg)(nil), &dkimTypes.MsgRemoveDkimPubKey{})
+
+	fundAmount := math.NewInt(10_000_000_000)
+	users := interchaintest.GetAndFundTestUsers(t, ctx, "default", fundAmount, xion)
+	chainUser := users[0]
+	govModAddress := GetModuleAddress(t, xion, ctx, govModule.ModuleName)
+
+	// query chain for DKIM records
+	dkimRecord, err := ExecQuery(t, ctx, xion.GetNode(), "dkim", "dkim-pubkey", domain_1, selector_1)
+	require.NoError(t, err)
+	require.Equal(t, dkimRecord["dkim_pubkey"].(map[string]interface{})["pub_key"].(string), pubKey_1)
+
+	// generate a dkim record by querying the chain
+	// and then submit a proposal to add it
+	dkimRecord, err = ExecQuery(t, ctx, xion.GetNode(), "dkim", "gdkim", "account.netflix.com", "kk6c473czcop4fqv6yhfgiqupmfz3cm2")
+	require.NoError(t, err)
+
+	customDkimPubkey := dkimRecord["pub_key"].(string)
+	poseidonHash, err := base64.StdEncoding.DecodeString(dkimRecord["poseidon_hash"].(string))
+	require.NoError(t, err)
+
+	governancePubKeys := []dkimTypes.DkimPubKey{
```

Comment:
Seems like you could use the `generate-dkim-pubkey` cmd for this

Reply:
Yes I did 
but the response from the endpoint does not marshal well into the struct because of the difference in encoding of enums so I just created the struct myself